gobblin-metrics/src/main/java/gobblin/metrics/reporter/ScheduledReporter.java

```diff
@@ -93,18 +93,14 @@ public static Config setReportingInterval(Config config, long reportingInterval,
     return config.withValue(REPORTING_INTERVAL, ConfigValueFactory.fromAnyRef(seconds + "S"));
   }
 
-  private final ScheduledExecutorService executor;
-  private final MetricFilter metricFilter;
+  private ScheduledExecutorService executor;
+  private MetricFilter metricFilter;
 
   private Optional<ScheduledFuture> scheduledTask;
   private int reportingPeriodSeconds;
 
   public ScheduledReporter(String name, Config config) {
     super(name, config);
-    this.executor = Executors.newSingleThreadScheduledExecutor(
-        ExecutorsUtils.newDaemonThreadFactory(Optional.of(log), Optional.of("metrics-" + name + "-scheduler")));
-    this.reportingPeriodSeconds = parsePeriodToSeconds(
-        config.hasPath(REPORTING_INTERVAL) ? config.getString(REPORTING_INTERVAL) : DEFAULT_REPORTING_INTERVAL_PERIOD);
     this.metricFilter = createMetricFilter(this.config);
```

Comment:
Can we move the initialization of this to `startImpl` I believe it will be used by the `report` method which is invoked by the thread pool.